core/src/main/scala/org/bitcoins/core/protocol/transaction/Transaction.scala

```diff
@@ -175,14 +164,23 @@ sealed abstract class WitnessTransaction extends Transaction {
 
 object Transaction extends Factory[Transaction] {
 
-  def fromBytes(bytes: ByteVector): Transaction = {
-    val wtxTry = Try(RawWitnessTransactionParser.read(bytes))
-    wtxTry match {
-      case Success(wtx) =>
-        wtx
-      case Failure(_) =>
-        val btx = RawBaseTransactionParser.read(bytes)
-        btx
+  override def fromBytes(bytes: ByteVector): Transaction = {
+    //see BIP141 for marker/flag bytes
+    //https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#transaction-id
+    if (bytes(4) == WitnessTransaction.marker && bytes(5) == WitnessTransaction.flag) {
+      //this throw/catch is _still_ necessary for the case where we have unsigned base transactions
+      //with zero inputs and 1 output which is serialized as "0001" at bytes 4 and 5.
+      //these transactions will not have a script witness associated with them making them invalid
+      //witness transactions (you need to have a witness to be considered a witness tx)
+      //see: https://github.com/bitcoin-s/bitcoin-s/blob/01d89df1b7c6bc4b1594406d54d5e6019705c654/core-test/src/test/scala/org/bitcoins/core/protocol/transaction/TransactionTest.scala#L88
+      try {
```

Comment:
as this try stuff only applies on base txs with no inputs and one output, can't we explicitly check 8 bytes ahead for a cmpct size uint (8 bytes ahead because we want to skip the satoshi value in the output) and then check to see if the size of the tx is `5 + currency_unit_size + cmpct_uint_size + cmpct_uint_result + lock_time_size` and if it is then interpret it as a base transaction, otherwise a witness transaction: no `try` necessary?

Reply:
https://github.com/bitcoin-s/bitcoin-s/issues/1442#issue-620451122